apps/web/components/RegisterMember.tsx

```diff
@@ -109,123 +78,100 @@ export function RegisterMember({
-    {
-      transaction: "Approve token expenditure",
-      message: "waiting for signature",
-      current: true,
-      dataContent: "1",
-      loading: false,
-      stepClassName: "idle",
-      messageClassName: "",
-    },
-    {
-      transaction: "Register",
-      message: "waiting for approval",
-      dataContent: "2",
-      current: false,
-      stepClassName: "idle",
-      messageClassName: "",
+  const {
+    write: writeRegisterMember,
+    transactionStatus: registerMemberTxStatus,
+  } = useContractWriteWithConfirmations({
+    ...registryContractCallConfig,
+    functionName: "stakeAndRegisterMember",
+    showNotification: false,
+    onConfirmations: () => {
+      publish({
+        topic: "member",
+        type: "add",
+        containerId: communityAddress,
+        function: "stakeAndRegisterMember",
+        id: communityAddress,
+        urlChainId,
+      });
     },
-  ];
+  });
+
+  const { allowanceTxProps: allowanceTx, handleAllowance } = useHandleAllowance(
+    accountAddress,
+    token.id as Address,
+    token.symbol,
+    communityAddress as Address,
+    registrationCost,
+    writeRegisterMember,
+  );
+
+  const [registrationTx, setRegistrationTx] = useState<TransactionProps>(
+    () => ({
+      contractName: `Register in ${communityName}`,
+      message: getTxMessage("idle"),
+      status: "idle",
+    }),
+  );
+
+  useEffect(() => {
+    setRegistrationTx((prev) => ({
+      ...prev,
+      message: getTxMessage(registerMemberTxStatus),
+      status: (registerMemberTxStatus as ComputedStatus) ?? "idle",
+    }));
+  }, [registerMemberTxStatus, communityName]);
+
+  const handleClick = () => {
+    if (isMember) {
+      writeUnregisterMember();
+    } else {
+      setIsOpenModal(true);
+      setRegistrationTx((prev) => ({
+        ...prev,
+        message: getTxMessage("idle"),
+        status: "idle",
+      }));
+      handleAllowance();
+    }
+  };
 
   return (
     <>
       <TransactionModal
-        ref={modalRef}
-        label="Register in community"
-        allowTokenStatus={allowTokenStatus}
-        stepTwoStatus={registerMemberStatus}
-        initialTransactionSteps={InitialTransactionSteps}
-        token={tokenSymbol}
-        pendingAllowance={pendingAllowance}
-        setPendingAllowance={setPendingAllowance}
+        label={`Register in ${communityName}`}
+        transactions={[allowanceTx, registrationTx]}
+        onClose={() => setIsOpenModal(false)}
+        isOpen={isOpenModal}
       />
       <div className="flex gap-4">
         <div className="flex items-center justify-center">
           <Button
             onClick={handleClick}
             btnStyle={isMember ? "outline" : "filled"}
             color={isMember ? "danger" : "primary"}
-            disabled={missmatchUrl || disabledRegMemberButton}
+            disabled={
+              missmatchUrl ||
+              disableRegMemberBtnCondition.some((cond) => cond.condition)
```

Comment:
Should be a useMemo I think because its can be costly to compute each time the component should rerender

Reply:
And since we cant assume the condition are not heavy computing

Reply:
fixed!